evalml/tests/automl_tests/test_automl.py

```diff
@@ -4488,6 +4464,77 @@ def test_cv_validation_scores_time_series(
     assert cv_vals[0] == validation_vals[0]
 
 
+@pytest.mark.parametrize("algorithm,batches", [("iterative", 2), ("default", 3)])
+@pytest.mark.parametrize(
+    "parameter,expected",
+    [
+        ("mean", ["mean", "median", "most_frequent"]),
+        (Categorical(["mean"]), Categorical(["mean"])),
+    ],
+)
+@pytest.mark.parametrize("problem_type", ["binary", "time series binary"])
+def test_search_parameters_held_automl(
+    problem_type, parameter, expected, algorithm, batches, X_y_binary, ts_data_binary
+):
+    if problem_type == "binary":
+        X, y = X_y_binary
+        problem_configuration = None
+        allowed_component_graphs = {
+            "cg": {
+                "Imputer": ["Imputer", "X", "y"],
+                "Label Encoder": ["Label Encoder", "Imputer.x", "y"],
+                "Decision Tree Classifier": [
+                    "Decision Tree Classifier",
+                    "Label Encoder.x",
+                    "Label Encoder.y",
+                ],
+            }
+        }
+    else:
+        X, y = ts_data_binary
+        problem_configuration = {
+            "time_index": "date",
+            "gap": 0,
+            "max_delay": 0,
+            "forecast_horizon": 1,
+        }
+        allowed_component_graphs = {
```

Comment:
@bchen1116 Do we have coverage that the pipeline hyperparameter search space is not constrained to be only the spaces in `search_parameters`? Can't find it.

Reply:
Can add!